### PR TITLE
DSpace 6.x - replace log4j with reload4j, slf4j-log4j12 with slf4j-reload4j

### LINKED ISSUE #8143: Refactor DSpace 6.x to use reload4j, slf4j-reload4j (log4j12 replacements with active security maintenance)
@misilot reported a new drop-in replacement for log4j 1.2.*, created by the original author of log4j 1.x, which is actively maintained for security and critical fixes.

I have a branch successfully building and running. Would love some help with testing! PR at  #8144 

I'm adding this as a public security / feature issue as this is not an active vulnerability, just some proactive maintenance related to security which we could do for future 6.x releases. I've tentatively added a milestone of 6.5 to avoid stalling 6.4 but if others are happy with it, perhaps we can squeeze it in for 6.4?

The project is called reload4j: https://reload4j.qos.ch
SLF4J has also published a replacement for slf4j-log4j12 called slf4j-reload4j: https://www.slf4j.org/news.html

These are meant to work **exactly** like log4j 1.2 and use the same java package path, so no logging code or import statements need to be changed, just dependency and exclusion configuration in our Maven POMs.

As noted when the log4j 2.x security vulnerabilities were being discussed and mitigated in many Java projects, DSpace versions prior to v7 were not affected as they use log4j 1.2.* (1.2.14-1.2.17 depending on exact transitive dependencies).
Although not vulnerable to the latest critical issue, there are several other security vulnerabilities out there which affect log4j 1.2.* and it does not get actively maintained. There will likely be more vulnerabilities discovered in the future.

reload4j and slf4j-reload4j were created to address the need for an exact replacement of log4j 1.2.17 with no changes apart from security maintenance.

A full list of the CVE references to fixed issues in reload4j can be found [at its webpage](https://reload4j.qos.ch/index.html).

### PR DESCRIPTION
**Pull request by kshepherd**
## References
* Fixes #8143 

## Description
See #8143 and https://reload4j.qos.ch for more information about reload4j

* Replaced log4j dependencies with reload4j 1.2.18 in `pom.xml` files for all relevant maven projects
* Replaced slf4j-log4j12 dependencies with slf4j-reload4j in `pom.xml` files for all relevant maven projects
* Wrote exclusions for some dependencies to ensure there are no transitive inclusions of the old log4j or slf4j-log4j12 artifacts anywhere in DSpace and to force usage of the new artifacts

No code or import changes are needed as these artifacts are exact replacements, written by the original log4j 1.x author, with a handful of security fixes included as per the [website](https://reload4j.qos.ch)

## Instructions for Reviewers

1. Build, preferably with `-U` and `clean` options so we can ensure a fresh build using reload4j
2. Basic application testing, ensuring every DSpace module / webapp / API can log normally, in exactly the same way as log4j 1.2.17
3. Test log4j configuration changes, ensure they are honoured after a restart
4. Review of POM changes - happy to hear suggestions if there are tidier ways to achieve all the transitive exclusions, etc.

### CODE REVIEW

**Comment by misilot** [comment]
I tested this locally on our dev instance we are in the process of standing up, and the custom logging locations appeared to work as expected.

Thank you @kshepherd for putting this together! This will make our security team on campus very happy.

File: pom.xml [hunk 1]
```diff
@@ -431,7 +431,7 @@
                 <plugins>
                     <plugin>
                         <groupId>org.codehaus.mojo</groupId>
-                        <artifactId>license-maven-plugin</artifactId>
+                        <artifactId>#license-maven-plugin</artifactId>
```
Comment (tdonohue) [inline pom.xml:434]: @kshepherd : Is this a typo?  Not sure what the `#` sign is doing here

**Review by tdonohue — COMMENTED** [review]
@kshepherd : Gave this a really quick code review (no test though, as I don't have a 6.x environment up right now).  It looks fine except for one place where there appears to be a typo?  See below.

File: pom.xml (hunk 1 above)
Comment (kshepherd) [reply pom.xml:434]: Thanks, not sure how that got there either, I'll update shortly

File: pom.xml [hunk 2]
```diff
@@ -32,6 +32,8 @@
         <solr.version>4.10.4</solr.version>
         <jena.version>2.13.0</jena.version>
         <slf4j.version>1.7.14</slf4j.version>
+        <reload4j.version>1.2.18.3</reload4j.version>
```
Comment (misilot) [inline pom.xml:35]: Should we target the latest release, 1.2.20?

Comment (kshepherd) [reply pom.xml:35]: I haven't had a chance to test that personally, but it sounds like a good idea

**Review by J4bbi — APPROVED** [review]
built, ran and changed to debug level, all good

**Review by nwoodward — APPROVED** [review]
👍 Deployed locally and the logs look good.

File: pom.xml [hunk 3]
```diff
@@ -32,6 +32,8 @@
...
+        <slf4j-reload4j.version>1.7.15</slf4j-reload4j.version>
```
Comment (alanorth) [inline pom.xml:36]: I think this was supposed to be `1.7.35`?

Comment (J4bbi) [reply pom.xml:36]: @kshepherd , should we bump of version(s)?

Comment (alanorth) [reply pom.xml:36]: @J4bbi I wasn't meaning to ask about bumping it. There is an error in the last patch that removes the hardcoded version `1.7.35` and replaces it with version `1.7.15`, with the intent of using a variable.

Comment (J4bbi) [reply pom.xml:36]: good catch, got your comment mixed up with discussion above.

right, so the value here should be `1.7.35` while the value in line 1363 below should be `${slf4j-reload4j.version}`?

File: pom.xml (hunk 2 above)
Comment (kshepherd) [reply pom.xml]: done!

File: pom.xml (hunk 3 above)
Comment (kshepherd) [reply pom.xml]: i'm not averse to bumping the version as well, i've done both - used variable and bumped slf4j-reload4j and reload4j to latest stable versions. all seems to be working well.

**Pull request merged**